derive_stringset/src/lib.rs

```diff
@@ -6,7 +6,16 @@ use syn::{parse_macro_input, DeriveInput};
 pub fn derive(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
 
-    let name = input.ident;
+    match &input.data {
+        syn::Data::Struct(_) => derive_struct(&input),
+        syn::Data::Enum(_) => derive_enum(&input),
```

Comment:
Why not pass the internals of the syn::Data::XX(data)` directly into the respective derive functions instead?

Reply:
Yeah, it was because of the name, I was debating between this and what you ended up doing by passing the name separately.